src/components/layout/nav/search/Search.jsx

```diff
@@ -1,7 +1,156 @@
+      "https://fgl.scene7.com/is/image/FGLSportsLtd/FGL_332762662_10_a-Nike-Womens-Air-Precision-II-Basketball-Shoes-White-Black-AA7069-100?bgColor=0,0,0,0&fmt=png-alpha&hei=528&resMode=sharp&qlt=85,1&op_sharpen=1",
+    ],
+  },
+];
 
 const Search = () => {
-  return <div></div>;
+  // It is left empty because we want search to start as empty.
+  const [searchQueryContainer, setSearchQuery] = useState("");
+  const [showEmptySearchWarning, setEmptySearchWarning] = useState(false);
+
+  const onlyFirstImageOfProduct = 0;
+  const amountOfProductToShowInSearch = 3;
+
+  // This will be sending input value to database with fetching if input is not empty when "Search" button is clicked.
+  const sendSearchQuery = () => {
+    if (!searchQueryContainer) {
+      setEmptySearchWarning(true);
+      return;
+    } else {
+      setEmptySearchWarning(false);
+    }
+  };
+
+  // This will setting input values onChange and will show preview card only if there is something typed.
+  const handleSearchChanges = (e) => {
+    setSearchQuery(e.target.value);
+    if (!searchQueryContainer) {
+      setEmptySearchWarning(true);
+      return;
+    } else {
+      setEmptySearchWarning(false);
+    }
+  };
+
+  const handleSearchPreview = () => {
+    const previewItemsContainer = mockDatabase
+      .filter((product) => product.tags.includes(searchQueryContainer))
+      .slice(0, amountOfProductToShowInSearch)
+      .map((product, idx) => (
+        <Col xl={3} lg={3} md={3} className="card" key={idx}>
+          <img src={product.images[onlyFirstImageOfProduct]} alt="product" />
+          <div className="title">{product.title}</div>
+          <div className="brand">{product.brand}</div>
+          <div className="price">
+            {product.currency}
+            {product.price}
+          </div>
+        </Col>
+      ));
+
+    // This will be more precise to show if it's still loading or failed as searching when we implement Fetching.
+    if (previewItemsContainer.lenght === 0) {
+      return <span>It's still loading.</span>;
+    } else {
+      return previewItemsContainer;
+    }
+  };
+
+  return (
+    <>
+      <Col className="searchBox col-12 searchTransition">
+        <Col xl={6} lg={7} md={8} className="searchBar">
+          <Col xl={10} lg={10} md={10} className="col-9 searchInputWrapper">
+            <Form.Control
+              size="lg"
+              type="text"
+              placeholder="Live search..."
+              className="searchInput"
+              onChange={handleSearchChanges}
+              value={searchQueryContainer}
+            />
+          </Col>
+          <Col xl={2} lg={2} md={2} className="col-3 searchButtonWrapper">
+            <Button onClick={sendSearchQuery}>Search</Button>
+          </Col>
+        </Col>
+        {searchQueryContainer ? (
+          <Col xl={6} lg={7} md={8} className="searchPreview">
+            {handleSearchPreview()}
```

Comment:
see comments on omar's code about this one (the PR he submitted)

usually we use handle for something like an onclick event

this is simply a variable that you can assign to some JSX, you don't need to make it a function
`const searchPreview = (<div></div>)` mesela